flood_adapt/dbs_classes/dbs_template.py

```diff
@@ -0,0 +1,271 @@
+
+        Raises
+        ------
+        ValueError
+            Raise error if name is already in use.
+        """
+
+        object_exists = object_model.attrs.name in self.list_objects()["name"]
+
+        # If you want to overwrite the object, and the object already exists, first delete it. If it exists and you
+        # don't want to overwrite, raise an error.
+        if overwrite and object_exists:
+            self.delete(object_model.attrs.name, toml_only=True)
+        elif not overwrite and object_exists:
+            raise ValueError(
+                f"'{object_model.attrs.name}' name is already used by another {self._type}. Choose a different name"
+            )
+
+        # If the folder doesnt exist yet, make the folder and save the object
+        if not (self._path / object_model.attrs.name).exists():
+            (self._path / object_model.attrs.name).mkdir()
+
+        object_model.save(
+            self._path / object_model.attrs.name / f"{object_model.attrs.name}.toml"
+        )
+
+    def edit(self, object_model: ObjectModel):
+        """Edits an already existing object in the database.
+
+        Parameters
+        ----------
+        object : ObjectModel
+            object to be edited in the database
+
+        Raises
+        ------
+        ValueError
+            Raise error if name is already in use.
+        """
+        # Check if the object exists
+        if object_model.attrs.name not in self.list_objects()["name"]:
+            raise ValueError(
+                f"'{object_model.attrs.name}' {self._type} does not exist. You cannot edit an {self._type} that does not exist."
+            )
+
+        # Check if it is possible to delete the object by saving with overwrite. This then
+        # also covers checking whether the object is a standard object, is already used in
+        # a higher level object. If any of these are the case, it cannot be deleted.
+        self.save(object_model, overwrite=True)
+
+    def delete(self, name: str, toml_only: bool = False):
+        """Deletes an already existing object in the database.
+
+        Parameters
+        ----------
+        name : str
+            name of the object to be deleted
+        toml_only : bool, optional
+            whether to only delete the toml file or the entire folder. If the folder is empty after deleting the toml,
+            it will always be deleted. By default False
+
+        Raises
+        ------
+        ValueError
+            Raise error if object to be deleted is already in use.
+        """
+        # Check if the object is a standard object. If it is, raise an error
+        if self._check_standard_objects(name):
+            raise ValueError(
+                f"'{name}' cannot be deleted/modified since it is a standard {self._type}."
+            )
+
+        # Check if object is used in a higher level object. If it is, raise an error
+        if used_in := self._check_higher_level_usage(name):
```

Comment:
cool way to do if statements, had to look it up to see if it wasnt a typo, but nice one